urleb43757003_Origraena_mas_platform_tgz-pJ8-ori_MainJ8/src/ori/mas/core/Properties.java

```diff
@@ -0,0 +1,69 @@
+package ori.mas.core;
+
+// property name => typed
+// 		<type>:<name>[-min|-max]
+public class Properties {
+
+	public static final String SUFFIX_MIN = "-min";
+	public static final String SUFFIX_MAX = "-max";
+
+	/** Used for agent conceptor recognition... */
+	// String
+	public static final String SOUL       = "soul";
+
+	// int
+	public static final String HEALTH     = "health";
+	public static final String HEALTH_MIN = HEALTH+SUFFIX_MIN;
+	public static final String HEALTH_MAX = HEALTH+SUFFIX_MAX;
+	// int
+	public static final String FEED       = "feed";
+	public static final String FEED_MIN   = FEED+SUFFIX_MIN;
+	public static final String FEED_MAX   = FEED+SUFFIX_MAX;
+
+	public static final String FEED_TICK  = "feed-tick"; // -1
+
+	// Body
+	public static final String TARGET     = "target";
+
+
+	// Convenient static methods
+	public static final Body getTarget(Body b) {
+		return (Body)(b.get(TARGET));
+	}
+	public static final int getHealth(Body b) {
+		try { return ((Integer)(b.get(HEALTH))).intValue(); }
+		catch (Exception e) { return 0; }
+	}
+	public static final int getHealthMax(Body b) {
+		try { return ((Integer)(b.get(HEALTH_MAX))).intValue(); }
+		catch (Exception e) { return 0; }
+	}
+	public static final int getFeed(Body b) {
+		try { return ((Integer)(b.get(FEED))).intValue(); }
+		catch (Exception e) { return 0; }
+	}
+	public static final int getFeedMax(Body b) {
+		try { return ((Integer)(b.get(FEED_MAX))).intValue(); }
+		catch (Exception e) { return 0; }
+	}
+	public static final float healthRatio(Body b) {
+		try {
+		float healthMax = (float)(((Integer)(b.get(HEALTH_MAX))).intValue());
+		float health = (float)(((Integer)(b.get(HEALTH))).intValue());
+		return health / healthMax;
+		}
+		catch (Exception e) { return 0.f; }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 50:55, 58:63.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.